### DISCUSSION TITLE
Login with SAML produces nondeterministic results and when successful redirects to login page

### DISCUSSION THREAD

**Discussion by Bloomby** [requarks/wiki#3722] [Error / Bug Report]
**Describe the bug**
Upon a successful login attempt with a SAML connection, the user is redirected back to the login page.

**To Reproduce**
1. Make sure you aren't logged in and go to the home link for our instance of the wiki
2. Click on the login button in the top right corner
3. Click the "SAML 2.0" button

1/3 of the time:
4. Gives Invalid authentication provider error.

1/3 of the time:
4. Takes me to internal company SAML authentication provider
5. I login with my company credentials
6. Ends at callback URL and gives Invalid authentication provider error.

1/3 of the time:
4. Takes me to internal company SAML authentication provider
5. I login with my company credentials
6. Successfully logs in but redirects the login page

**Expected behavior**
100% of the time SAML button connects successfully to my auth provider, and on a successful login send me to the home page of the wiki, not redirect me to the login page.

**Host Info (please complete the following information):**
 - OS: Docker Image direct from the Docker Repo running in an azure web app
 - Wiki.js version: 2.0.12
 - Database engine: postgress 11

**Additional context**
I have added the Read rule proposed in other issues of this matter but that did not make a difference (see screen shot)
<img width="1299" alt="Screen Shot 2020-02-10 at 4 19 21 PM" src="https://user-images.githubusercontent.com/25759715/74190975-3086e500-4c21-11ea-9a55-5a931bb736be.png">

**Comment by sphen13** [comment]
Thanks for filing this. I see this as well which ends up confusing users as they dont get to where i am trying to send them. any update on this?

**Comment by sphen13** [comment]
FYI i am running WIki.js 2.5.170 on docker.
